rcl/src/rcl/node.c

```diff
@@ -387,6 +388,20 @@ rcl_node_fini(rcl_node_t * node)
       result = RCL_RET_ERROR;
     }
   }
+  rcl_ret = rcl_node_type_description_service_fini(node);
+  if (rcl_ret == RCL_RET_NOT_INIT) {
+    rcl_reset_error();
+  } else if (rcl_ret != RCL_RET_OK) {
+    RCL_SET_ERROR_MSG("Unable to fini ~/get_type_description service for node.");
+    result = RCL_RET_ERROR;
+  }
```

Comment:
It seems somewhat asymmetric to `fini` the service here when we didn't `init` the service in `rcl_node_init`.  Do we actually need to call this here, or should we leave it to the client libraries to do this?

Reply:
I suppose that `rmw_destroy_node` will destroy the service anyways